paymentsheet/src/test/java/com/stripe/android/paymentsheet/IntentConfirmationFlowTest.kt

```diff
@@ -0,0 +1,272 @@
+        return PaymentConfirmationOption.PaymentMethod.New(
+            initializationMode = PaymentSheet.InitializationMode.DeferredIntent(
+                intentConfiguration = PaymentSheet.IntentConfiguration(
+                    mode = PaymentSheet.IntentConfiguration.Mode.Setup(
+                        currency = "USD",
+                    )
+                )
+            ),
+            createParams = PaymentMethodCreateParamsFixtures.DEFAULT_CARD,
+            optionsParams = null,
+            shouldSave = false,
+            shippingDetails = AddressDetails(
+                name = "John Doe",
+                phoneNumber = "1234567890"
+            ),
+        )
+    }
+
+    private fun createIntentConfirmationDefinition(
+        createPaymentMethodResult: Result<PaymentMethod> = Result.success(CARD_PAYMENT_METHOD),
+        intentResult: Result<StripeIntent> =
+            Result.success(SetupIntentFixtures.SI_REQUIRES_PAYMENT_METHOD),
+    ): IntentConfirmationDefinition {
+        return IntentConfirmationDefinition(
+            intentConfirmationInterceptor = DefaultIntentConfirmationInterceptor(
+                isFlowController = false,
+                publishableKeyProvider = {
+                    "pk_123"
+                },
+                stripeAccountIdProvider = {
+                    "acct_123"
+                },
+                stripeRepository = FakeStripeRepository(
+                    createPaymentMethodResult = createPaymentMethodResult,
+                    retrieveIntent = intentResult,
+                ),
+            ),
+            paymentLauncherFactory = {
+                FakePaymentLauncher()
+            }
+        )
+    }
+
+    private fun ConfirmStripeIntentParams.asSetup(): ConfirmSetupIntentParams {
+        return this as ConfirmSetupIntentParams
+    }
+
+    private fun IntentConfirmationDefinition.Args.asConfirm(): IntentConfirmationDefinition.Args.Confirm {
+        return this as IntentConfirmationDefinition.Args.Confirm
+    }
+
+    private fun <TLauncherArgs> PaymentConfirmationDefinition.ConfirmationAction<TLauncherArgs>.asFail():
+        PaymentConfirmationDefinition.ConfirmationAction.Fail<TLauncherArgs> {
+        return this as PaymentConfirmationDefinition.ConfirmationAction.Fail<TLauncherArgs>
+    }
+
+    private fun <TLauncherArgs> PaymentConfirmationDefinition.ConfirmationAction<TLauncherArgs>.asComplete():
+        PaymentConfirmationDefinition.ConfirmationAction.Complete<TLauncherArgs> {
+        return this as PaymentConfirmationDefinition.ConfirmationAction.Complete<TLauncherArgs>
+    }
+
+    private fun <TLauncherArgs> PaymentConfirmationDefinition.ConfirmationAction<TLauncherArgs>.asLaunch():
+        PaymentConfirmationDefinition.ConfirmationAction.Launch<TLauncherArgs> {
+        return this as PaymentConfirmationDefinition.ConfirmationAction.Launch<TLauncherArgs>
+    }
+}
```

Comment:
I was hoping to have a more integrated style of test for this where we create an activity and view model that generate Dagger test components that use the same modules as `PaymentSheet` and `FlowController` whose sole function is intent confirmation. I could not get Dagger to generate those test components in the `test` directory. I'll look deeper into it in a follow-up for this PR.